pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -257,6 +260,43 @@ func (r *InferenceGraphReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		}
 	}
 
+	// Handle InferenceGraph status updates based on the force stop annotation.
+	// If true, transition the service to a stopped and unready state; otherwise, ensure it's not marked as stopped.
+	transition_time := apis.VolatileTime{Inner: metav1.Now()}
+	existingStoppedCondition := graph.Status.GetCondition(v1beta1.Stopped)
+	if forceStopRuntime {
+		// If the graph's stopped condition is not set or
+		// If the graph is currently running, update its status to signal that it should be stopped
+		if existingStoppedCondition == nil || (existingStoppedCondition != nil && existingStoppedCondition.Status == corev1.ConditionFalse) {
```

Comment:
The condition:
```
if existingStoppedCondition == nil || (existingStoppedCondition != nil && existingStoppedCondition.Status == corev1.ConditionFalse) {
```
contains a duplicated existingStoppedCondition != nil check.

Since the first clause (existingStoppedCondition == nil) already ensures safety, the second clause doesn’t need an explicit nil check.

It can be simplified to:
```
if existingStoppedCondition == nil || existingStoppedCondition.Status == corev1.ConditionFalse {
```
This improves readability without changing the logic.